overlord/snapstate/snapstate.go

```diff
@@ -597,8 +597,13 @@ func validateFeatureFlags(st *state.State, info *snap.Info) error {
 	return nil
 }
 
-func checkInstallPreconditions(st *state.State, info *snap.Info, flags Flags, snapst *SnapState, deviceCtx DeviceContext) error {
-	if err := validateInfoAndFlags(info, snapst, flags); err != nil {
+func ensureInstallPreconditions(st *state.State, info *snap.Info, flags *Flags, snapst *SnapState, deviceCtx DeviceContext) error {
```

Comment:
my slight preference would be for this to return the mutated flags instead of mutating in place

Reply:
actually not slight, if ones passes &snapst.Flags we could get behavior that is not quite right vs undos

Reply:
mmh, likely not undo, but we do some delicate things with snapst in snapstate.go that don't expect it to change under their back, so really prefer to be careful here